acceptance/suites/puppet3_tests/020_backwards_compat/test.rb

```diff
@@ -0,0 +1,124 @@
+  end
+end
+
+step "Validate content-file" do
+  contents = on(agent, "cat #{studio}/content-file").stdout.chomp
+  assert_equal('Static content defined in manifest', contents)
+end
+
+step "Validate source-file" do
+  contents = on(agent, "cat #{studio}/source-file").stdout.chomp
+  assert_equal('Static source file contents', contents)
+end
+
+step "Validate source-file filebucket backup" do
+  old_md5 = on(agent, "openssl dgst -md5 #{studio}/source-file-old | awk '{print $2}'").stdout.chomp
+  on(agent, puppet("filebucket restore #{studio}/source-file-backup #{old_md5} --server #{agent}"))
+  diff = on(agent, "diff #{studio}/source-file-old #{studio}/source-file-backup").exit_code
+  assert_equal(0, diff, 'source-file was not backed up to filebucket')
+end
+
+step "Validate binary-file" do
+  md5 = on(agent, "openssl dgst -md5 #{studio}/binary-file | awk '{print $2}'").stdout.chomp
+  assert_equal('7cfc2db80222ef224d99648716cea8e4', md5)
+end
+
+step "Validate binary-file filebucket backup" do
+  old_md5 = on(agent, "openssl dgst -md5 #{studio}/binary-file-old | awk '{print $2}'").stdout.chomp
+  on(agent, puppet("filebucket restore #{studio}/binary-file-backup #{old_md5} --server #{agent}"))
+  diff = on(agent, "diff #{studio}/binary-file-old #{studio}/binary-file-backup").exit_code
+  assert_equal(0, diff, 'binary-file was not backed up to filebucket')
+end
+
+step "Validate recursive-directory" do
+  directory_exists = on(agent, "test -d #{studio}/recursive-directory").exit_code
+  assert_equal(0, directory_exists)
+
+  file1_contents = on(agent, "cat #{studio}/recursive-directory/file1").stdout.chomp
+  assert_equal('recursive file 1', file1_contents)
+
+  file2_contents = on(agent, "cat #{studio}/recursive-directory/file2").stdout.chomp
+  assert_equal('recursive file 2', file2_contents)
+
+  subdir_exists = on(agent, "test -d #{studio}/recursive-directory/subdir").exit_code
+  assert_equal(0, subdir_exists)
+
+  subfile_contents = on(agent, "cat #{studio}/recursive-directory/subdir/subfile").stdout.chomp
+  assert_equal('recursive subfile contents', subfile_contents)
+end
+
+step "Validate mount-point-source-file" do
+  contents = on(agent, "cat #{studio}/mount-point-source-file").stdout.chomp
+  assert_equal('File served from custom mount point', contents)
+end
+
+step "Validate mount-point-binary-file" do
+  md5 = on(agent, "openssl dgst -md5 #{studio}/mount-point-binary-file | awk '{print $2}'").stdout.chomp
+  assert_equal('4b392568e0c19886bf274663a63b7d18', md5)
+end
+
+step "Validate custom-fact-output" do
+  expected = on(agent, "hostname").stdout.chomp
+  content = on(agent, "cat #{studio}/custom-fact-output").stdout.chomp
+  assert_equal(expected, content)
+end
+
+step "Validate external-fact-output" do
+  expected = on(agent, "hostname").stdout.chomp
+  content = on(agent, "cat #{studio}/external-fact-output").stdout.chomp
+  assert_equal(expected, content)
```

Comment:
As discussed with @nwolfe, this test will fail on Puppet 4 because of a defect with external, executable facts not being set executable on the agent after having been downloaded.  We'll probably need to set this one to skip with a pointer to the PUP ticket against the agent that will fix this problem.